ee/clickhouse/client.py

```diff
@@ -131,16 +134,25 @@ def sync_execute(query, args=None, settings=None, with_column_types=False):
             if app_settings.SHELL_PLUS_PRINT_SQL:
                 print()
                 print(format_sql(query, args))
+
+            sql, tags = _annotate_tagged_query(query, args)
+            timeout_task = QUERY_TIMEOUT_THREAD.schedule(_notify_of_slow_query_failure, tags)
+
             try:
-                sql, tags = _annotate_tagged_query(query, args)
                 result = client.execute(sql, args, settings=settings, with_column_types=with_column_types)
             except Exception as err:
                 tags["failed"] = True
                 tags["reason"] = type(err).__name__
+                incr("clickhouse_sync_execution_failure", tags=tags)
+
                 raise _wrap_api_error(err)
             finally:
                 execution_time = time() - start_time
```

Comment:
Aside: https://docs.python.org/3/library/time.html#time.perf_counter - Perf_counter is better than time() for measuring times, specially for O(1 second) queries.